### PR TITLE
Setting up auto-release jobs for build, eventing and serving

### LINKED ISSUE #368: Automate the creation of major releases
Summary: a bi-hourly job (to ensure enough time for building a release) checks for new release branches (using the releases published on GitHub for comparison), then builds a new major release.

### PR DESCRIPTION
**Pull request by nbarthwal**
Setting up auto-release jobs for build, eventing and serving.

Fixes #368.

### CODE REVIEW

File: ci/prow/config.yaml [hunk 1]
```diff
@@ -1537,6 +1570,31 @@ periodics:
...
+    containers:
+    - image: gcr.io/knative-tests/test-infra/prow-tests:stable
+      imagePullPolicy: Always
+      args:
+      - "--scenario=kubernetes_execute_bazel"
+      - "--clean"
+      - "--job=$(JOB_NAME)"
+      - "--repo=github.com/knative/build"
+      - "--root=/go/src"
+      - "--service-account=/etc/nightly-account/service-account.json"
```
Comment (adrcunha) [inline ci/prow/config.yaml]:
```suggestion
      - "--service-account=/etc/release-account/service-account.json"
```

File: ci/prow/config.yaml [hunk 2]
```diff
@@ -1537,6 +1570,31 @@ periodics:
...
+      - "--upload=gs://knative-prow/logs"
+      - "--timeout=90" # 1.5h
+      - "--" # end bootstrap args, scenario args below
+      - "--" # end kubernetes_execute_bazel flags (consider following flags as text)
+      - "./hack/release.sh"
+      - "--auto-release"
```
Comment (adrcunha) [inline ci/prow/config.yaml]: Missing GCR, GCS and token. Remember to also include the token volume.

File: ci/prow/config.yaml [hunk 3]
```diff
@@ -1781,6 +1839,28 @@ periodics:
...
+      - "--job=$(JOB_NAME)"
+      - "--repo=github.com/knative/eventing"
+      - "--root=/go/src"
+      - "--service-account=/etc/release-account/service-account.json"
+      - "--upload=gs://knative-prow/logs"
+      - "--timeout=90" # 1.5h
+      - "--" # end bootstrap args, scenario args below
+      - "--" # end kubernetes_execute_bazel flags (consider following flags as text)
+      - "./hack/release.sh"
+      - "--auto-release"
```
Comment (adrcunha) [inline ci/prow/config.yaml]: Missing GCR, GCS and token. Remember to also include the token volume.

**Comment by adrcunha** [comment]
Also, before merging, make sure all 3 jobs run fine. Use `run_job.sh` for this.

File: ci/prow/config.yaml [hunk 4]
```diff
@@ -1537,6 +1570,42 @@ periodics:
       # docker needs privileged mode
       securityContext:
         privileged: true
+- cron: "7 */2 * * *" # Run every two hours and seven minutes
+  name: ci-knative-build-auto-release
+  agent: kubernetes
+  labels:
+    preset-release-account: "true"
```
Comment (adrcunha) [inline ci/prow/config.yaml]: Also add `preset-dind-enabled: "true"`

**Comment by adrcunha** [comment]
/lgtm
/hold

Unhold it when your jobs are ok.

**Comment by nbarthwal** [comment]
check-cla

**Comment by nbarthwal** [comment]
The logs for these tests are:
https://gubernator.knative.dev/build/knative-prow/pr-logs/pull/knative_test-infra/464/pull-knative-test-infra-build-tests/1095089382865506304/
https://gubernator.knative.dev/build/knative-prow/pr-logs/pull/knative_test-infra/464/pull-knative-test-infra-integration-tests/1095089382865506306/
https://gubernator.knative.dev/build/knative-prow/pr-logs/pull/knative_test-infra/464/pull-knative-test-infra-unit-tests/1095089382865506305/

**Comment by adrcunha** [comment]
This won't work. The code ran by the jobs is pull straight from HEAD.

**Comment by nbarthwal** [comment]
But can’t I run code from my HEAD of my branch?

I can manually do a run_tests.sh to run thesejobsin Prow

**Comment by nbarthwal** [comment]
https://gubernator.knative.dev/build/knative-prow/logs/ci-knative-build-auto-release/1095461264974417920/
https://gubernator.knative.dev/build/knative-prow/logs/ci-knative-eventing-auto-release/1095461264978612224/
https://gubernator.knative.dev/build/knative-prow/logs/ci-knative-build-auto-release/1095461264974417920/

**Comment by nbarthwal** [comment]
https://gubernator.knative.dev/build/knative-prow/logs/ci-knative-serving-auto-release/1095461140059656192/
https://gubernator.knative.dev/build/knative-prow/logs/ci-knative-eventing-auto-release/1095461264978612224/
https://gubernator.knative.dev/build/knative-prow/logs/ci-knative-build-auto-release/1095461264974417920/

**Pull request merged**